logstash-core/lib/logstash/plugins/builtin/pipeline/input.rb

```diff
@@ -55,16 +56,22 @@ def running?
   # To understand why this value is useful see Internal.send_to
   # Note, this takes a java Stream, not a ruby array
   def internalReceive(events)
-    return false if !@running.get()
+    return ReceiveResponse.closing() if !@running.get()
 
     # TODO This should probably push a batch at some point in the future when doing so
     # buys us some efficiency
-    events.forEach do |event|
-      decorate(event)
-      @queue << event
+    begin
+      stream_position = 0
+      events.forEach do |event|
+        decorate(event)
+        @queue << event
+        stream_position = stream_position + 1
+      end
+      ReceiveResponse.completed()
+    rescue ava.lang.InterruptedException, IOError => e
```

Comment:
```suggestion
    rescue java.lang.InterruptedException, IOError => e
```
we clearly done have this path tested, maybe a naive unit spec just doing a stubbed `internalReceive` would make sense?

Reply:
Yep added test! Good point